core/src/main/java/org/fao/geonet/kernel/mef/MEF2Exporter.java

```diff
@@ -323,23 +323,28 @@ private static void createMetadataFolder(ServiceContext context,
             Files.write(featureMdDir.resolve(FILE_METADATA), ftrecordAndMetadata.two().getBytes(CHARSET));
         }
 
+        final Store store = context.getBean("resourceStore", Store.class);
+        final List<MetadataResource> publicResources = store.getResources(context, metadata.getUuid(),
+                MetadataResourceVisibility.PUBLIC, null, true);
+        final List<MetadataResource> privateResources = store.getResources(context, metadata.getUuid(),
+                MetadataResourceVisibility.PRIVATE, null, true);
 
         // --- save info file
-        byte[] binData = MEFLib.buildInfoFile(context, record, format, pubDir,
-            priDir, skipUUID).getBytes(Constants.ENCODING);
+        byte[] binData = MEFLib.buildInfoFile(context, record, format, publicResources,
+                privateResources, skipUUID).getBytes(Constants.ENCODING);
 
         Files.write(metadataRootDir.resolve(FILE_INFO), binData);
 
         // --- save thumbnails and maps
 
         if (format == Format.PARTIAL || format == Format.FULL) {
-            IO.copyDirectoryOrFile(pubDir, metadataRootDir, true);
+            StoreUtils.extract(context, metadata.getUuid(), publicResources, zipFs.getPath("public"), true);
```

Comment:
It looks like this breaks the MEF format mixing all attachments in a root public and private folder instead of being dispatch by record.
![image](https://user-images.githubusercontent.com/1701393/73072196-ebd71c00-3eb4-11ea-9d49-2902ef75cc1d.png)